packages/pigeon/lib/swift_generator.dart

```diff
@@ -0,0 +1,642 @@
+            indent.scoped('', '', () {
+              indent
+                  .write('guard let map = self.readValue() as? [String: Any] ');
+              indent.scoped('else {', '}', () {
+                indent.writeln('return nil');
+              });
+              indent.write(
+                  'return ${_className(prefix, customClass.name)}.fromMap(map)');
+            });
+          }
+          indent.write('default:');
+          indent.scoped('', '', () {
+            indent.writeln('return super.readValue(ofType: type)');
+          });
+        });
+      });
+    }
+  });
+
+  // Generate Writer
+  indent.write('private class $writerName: FlutterStandardWriter ');
+  indent.scoped('{', '}', () {
+    if (getCodecClasses(api, root).isNotEmpty) {
+      indent.write('override func writeValue(_ value: Any) ');
+      indent.scoped('{', '}', () {
+        indent.write('');
+        for (final EnumeratedClass customClass in getCodecClasses(api, root)) {
+          indent.write(
+              'if let value = value as? ${_className(prefix, customClass.name)} ');
+          indent.scoped('{', '} else ', () {
+            indent.writeln('super.writeByte(${customClass.enumeration})');
+            indent.writeln('super.writeValue(value.toMap())');
+          }, addTrailingNewline: false);
+        }
+        indent.scoped('{', '}', () {
+          indent.writeln('super.writeValue(value)');
+        });
+      });
+    }
+  });
+  indent.writeln('');
+
+  // Generate ReaderWriter
+  indent.write('private class $readerWriterName: FlutterStandardReaderWriter ');
+  indent.scoped('{', '}', () {
+    indent.write(
+        'override func reader(with data: Data) -> FlutterStandardReader ');
+    indent.scoped('{', '}', () {
+      indent.writeln('return $readerName(data: data)');
+    });
+    indent.writeln('');
+    indent.write(
+        'override func writer(with data: NSMutableData) -> FlutterStandardWriter ');
+    indent.scoped('{', '}', () {
+      indent.writeln('return $writerName(data: data)');
+    });
+  });
+  indent.writeln('');
+
+  // Generate Codec
+  indent.write('class $codecName: FlutterStandardMessageCodec ');
+  indent.scoped('{', '}', () {
+    indent.writeln(
+        'static let shared = $codecName(readerWriter: $readerWriterName())');
+  });
+}
+
+/// Write the swift code that represents a host [Api], [api].
+/// Example:
+/// protocol Foo {
+///   Int add(x: Int, y: Int);
+///   static func setup(FlutterBinaryMessenger binaryMessenger, Foo api) {...}
+/// }
+void _writeHostApi(String? prefix, Indent indent, Api api) {
+  assert(api.location == ApiLocation.host);
+
+  final String apiName = _className(prefix, api.name);
+
+  indent.writeln(
+      '/** Generated protocol from Pigeon that represents a handler of messages from Flutter.*/');
+  indent.write('protocol $apiName ');
+  indent.scoped('{', '}', () {
+    for (final Method method in api.methods) {
```

Comment:
Really. Nice idea.
I think in an annotation in each parameter to define this.

 How about that?
```dart
@HostApi()
abstract class Api {
  String? getString(@ParameterLabel('for') String key);
  void setString(@ParameterLabel('_') String value, @ParameterLabel('for') String key);
  void remove(String key);
}
```

This would generate

### ObjC
```objc
@protocol Api
- (nullable NSString *)getStringForKey:(NSString *)key error:(FlutterError *_Nullable *_Nonnull)error;
/// @return `nil` only when `error != nil`.
- (void)setString:(NSString *)value forKey:(NSString *)key error:(FlutterError *_Nullable *_Nonnull)error;
/// @return `nil` only when `error != nil`.
- (void)removeKey:(NSString *)key error:(FlutterError *_Nullable *_Nonnull)error;
@end
```

### Swift
```swift
protocol Api {
  func getString(for key: String) -> String?
  func setString(_ value: String, for key: String)
  func remove(key: String)
}
```

This samples are about HostApi, but make more sense FlutterApi because the native code will consume.

Reply:
> I would expect it to work the same way as `@ObjCSelector` so that there aren't two completely different ways of doing essentially the same thing.

I've be thinking about this. How about we change this annotation name? (of course, duplicate and deprecate)
Maybe to `MethodSelector`, `Selector`?
Thus, it could be used both to ObjC and Swift.

Reply:
Yea, I thought about that.  I don't like those names since "selector" is objc specific.  You don't technically have one in Swift.

I don't want you to get bogged down with this decision.  Maybe for now we can just reuse `@ObjcSelector` and consider refactoring it in a different PR?  Or better yet, just don't support this yet and file an issue.  @stuartmorgan alright said it's fine to defer this for now too.  There is already a lot to review in this one PR.

Reply:
To be clear, I wasn't suggesting a single annotation that covers both. The naming conventions for ObjC and Swift are not always going to be the same, so they should be separately controllable. I just meant it shouldn't be param-based here, and method-based for ObjC. We should follow the pattern established by the ObjC annotation for the new Swift annotation.

Reply:
Ok, I got it.
I could support this annotation and open a file a new issue, but i dont know how proceed to create a issue to a feature that even not exists.
Should I wait this PR be merged to file a issue to discuss a new annotation to Swift?

Reply:
> I could support this annotation and open a file a new issue, but i dont know how proceed to create a issue to a feature that even not exists.

It's fine to open an issue that says something like "PR #XYZ will add Swift support, but [...]".

Reply:
Issue created: https://github.com/flutter/flutter/issues/105932